pppd/sys-linux.c

```diff
@@ -1596,11 +1596,11 @@ get_ppp_stats_rtnetlink(int u, struct pppd_stats *stats)
 
     if (nlresp.nlh.nlmsg_type == NLMSG_ERROR) {
 	if (nlresplen < offsetof(struct nlresp, __end_err)) {
-	    if (kernel_version >= KVERSION(4,7,0))
-		error("get_ppp_stats_rtnetlink: Netlink responded with error: %s (line %d)", strerror(-nlresp.nlerr.error), __LINE__);
-	} else {
 	    error("get_ppp_stats_rtnetlink: Netlink responded with an error message, but the nlmsgerr structure is incomplete (line %d).",
 		    __LINE__);
+	} else {
+	    if (kernel_version >= KVERSION(4,7,0))
```

Comment:
This can be } else if {
... no need for an additional layer of nesting.

If we know the errno on <= 4.7.0 perhaps we should rather test for that specicif errno and filter only that?

Reply:
I would not recommend to test on specific error returned by pre-4.7 kernels. Same errno code could be returned also in future for some other error. Codes are already too vague and used for many different errors (e.g. EINVAL).

Reply:
That's fair.  I'm happy.  My one comment then is style related and it's much of a muchness.